bgs/bgs.go

```diff
@@ -494,12 +554,12 @@ func (s *BGS) createExternalUser(ctx context.Context, did string) (*models.Actor
 	u := User{
 		Handle: handle,
 		Did:    did,
-		PDS:    peering.ID,
+		PDS:    pdsid,
 	}
 
 	if err := s.db.Create(&u).Error; err != nil {
-		// some debugging...
-		return nil, fmt.Errorf("failed to create other pds user: %w", err)
+		log.Warnf("failed to create new user record: %s", err)
+		return nil, ErrRevalidateUserHandle
```

Comment:
It isn't obvious to me why we would always try to re-validate if the row insertion failed. Eg, couldn't the database error be a deadlock or connection dropped or something other than a constraint violation? Feels like this needs something like the `if !errors.Is(err, gorm.ErrRecordNotFound)` just above.

Reply:
yeah, There didnt seem to be an easy way to check that when i looked through the gorm docs otherwise i would have

Reply:
From https://github.com/go-gorm/gorm/blob/master/errors.go, it looks like `ErrDuplicatedKey` would at least handle the "there is already such a handle" UNIQUE constraint violation. If that is the primary situation where the creation would fail and we need to re-validate.